tests/test_DGEList.py

```diff
@@ -203,11 +203,24 @@ def test_cpm():
     assert dge_list.counts[0][0] == first_pos * 1e6 / col_sum[0]
 
 
+def test_tpm():
+    dge_list = DGEList(filename=str(get_dataset_path(TEST_DATASET_NPZ)))
+    first_pos = dge_list.counts[0][0]
+    gene_lengths = np.random.randint(10, 1000, size=dge_list.genes.shape)
+    desired = (
+        1e6
+        * (first_pos / gene_lengths[0])
+        / np.sum(dge_list.counts / gene_lengths[:, None], axis=0)[0]
+    )
```

Comment:
Not bad, you are dynamically creating an expected result by copying some code from the main application.

This makes it hard to track down bugs with the logic since the logic exists in two places.

Norm. methods should probably be tested with a very small, hand-designed test case so you can explicitly write out all inputs, and all outputs.

Reply:
You are definitely right, that this test can be improved. I adopted it from cpm_test tbh. 
I'll improve by writing and calculating a "hand calculated simple test"!

Reply:
Done using hand written test and property validation